### DISCUSSION TITLE
Fortify integration error

### DISCUSSION THREAD

**Discussion by hdshiva** [DependencyTrack/dependency-track#701] [Q&A]
I have updated ssc integration user creds and but unable to see the integration going through to the application on my fortify. I'm using  Dependency-Track  v3.8.0 and fortify version is 18.20.1071
Could you please advise what could be wrong here. when try to curl api url `https://fortify.va.poc.it/ssc/api/v1` I get the success response with the credentials that updated in integration section of dependency-tracker.

> 2020-06-11 23:36:41,075 [] ERROR [org.dependencytrack.integrations.fortifyssc.FortifySscClient] An error occurred while communicating with the Fortify SSC integration point
> 2020-06-11 23:36:41,075 [] ERROR [org.dependencytrack.integrations.fortifyssc.FortifySscClient] HTTP Status : 401 
> 2020-06-11 23:36:41,075 [] ERROR [org.dependencytrack.integrations.fortifyssc.FortifySscClient] Request URL : https://fortify.va.poc.it/ssc/api/v1/fileTokens
> 2020-06-11 23:41:41,049 [] WARN [org.dependencytrack.integrations.fortifyssc.FortifySscClient] Fortify SSC Client did not receive expected response while attempting to generate a one-time-use fileupload token. HTTP response code: 401 - 
> 2020-06-11 23:41:41,049 [] ERROR [org.dependencytrack.integrations.fortifyssc.FortifySscClient] An error occurred while communicating with the Fortify SSC integration point
> 2020-06-11 23:41:41,049 [] ERROR [org.dependencytrack.integrations.fortifyssc.FortifySscClient] HTTP Status : 401 
> 2020-06-11 23:41:41,049 [] ERROR [org.dependencytrack.integrations.fortifyssc.FortifySscClient] Request URL : https://fortify.va.poc.it/ssc/api/v1/fileTokens

**Comment by stevespringett** [comment]
Is your SSC also at 18.20 or has it been upgraded to 20.1?

20.1 had breaking changes as detailed in #700

**Comment by hdshiva** [comment]
Our SSC is at 18.20.1071 not upgraded to 20.1.

**Comment by chrissike** [comment]
Hi @hdshiva and @stevespringett ,
I'm currently running into the same issue. My SSC installation is at 19.2.0.3191.

Could this have to do with the type of token I need to use in the password field of DependencyTrack?
base64 or fortify ssc client token? and do I need a specific SSC token type? or really just the password?

Thanks for any hint.
Have a nice day
BR Christina
____________________

Here's my output with DEBUG:

14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.impl.conn.DefaultManagedHttpClientConnection - http-outgoing-293: set socket timeout to 0
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.impl.conn.DefaultManagedHttpClientConnection - http-outgoing-293: set socket timeout to 60000
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.impl.execchain.MainClientExec - Executing request POST /api/v1/fileTokens HTTP/1.1
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.impl.execchain.MainClientExec - Proxy auth state: UNCHALLENGED
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 >> POST /api/v1/fileTokens HTTP/1.1
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 >> Content-Type: application/json
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 >> Authorization: Basic <basic auth string>=
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 >> user-agent: unirest-java/3.0.00
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 >> accept-encoding: gzip
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 >> Content-Length: 26
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 >> Host: my.ssc.portal
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 >> Connection: Keep-Alive
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 >> "POST /api/v1/fileTokens HTTP/1.1[\r][\n]"
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 >> "Content-Type: application/json[\r][\n]"
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 >> "Authorization: Basic  <basic auth string>=[\r][\n]"
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 >> "user-agent: unirest-java/3.0.00[\r][\n]"
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 >> "accept-encoding: gzip[\r][\n]"
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 >> "Content-Length: 26[\r][\n]"
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 >> "Host: my.ssc.portal[\r][\n]"
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 >> "Connection: Keep-Alive[\r][\n]"
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 >> "[\r][\n]"
14:05:51.043 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 >> "{"fileTokenType":"UPLOAD"}"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "HTTP/1.1 401 [\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "X-Frame-Options: DENY[\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "X-Content-Type-Options: nosniff[\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "X-XSS-Protection: 1; mode=block[\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "X-Permitted-Cross-Domain-Policies: none[\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "WWW-Authenticate: Basic realm="hp:fortify:ssc:api"[\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "Content-Encoding: gzip[\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "Content-Type: application/json[\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "Content-Length: 96[\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "Date: Fri, 17 Jul 2020 12:05:51 GMT[\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "[\r][\n]"
14:05:51.110 [Alpine-EventService-7] DEBUG org.apache.http.wire - http-outgoing-293 << "[0x1f][0x8b][0x8][0x0][0x0][0x0][0x0][0x0][0x0][0x0][0xab]V[0xca]M-.NLOU[0xb2]RrLN[0x6][0xb2][0x15]\R[0xf3]2SS[0xf4][0x14]B[0xf3][0x12]KK2[0xf2][0x8b]2[0xab]RS[0x14][0x12][0xc1]rJ:JE[0xa9][0xc5][0x5][0xf9]y[0xc5][0xa9][0xce][0xf9])@=&[0x6][0x86]:J[0xa9]EE[0xf9]E[0x10][0xbe][0xae][0xa1][0x81][0xb1][0x81]a-[0x0][0x9e][0x99]5/V[0x0][0x0][0x0]"
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 << HTTP/1.1 401
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 << X-Frame-Options: DENY
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 << X-Content-Type-Options: nosniff
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 << X-XSS-Protection: 1; mode=block
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 << X-Permitted-Cross-Domain-Policies: none
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 << WWW-Authenticate: Basic realm="hp:fortify:ssc:api"
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 << Content-Encoding: gzip
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 << Content-Type: application/json
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 << Content-Length: 96
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.headers - http-outgoing-293 << Date: Fri, 17 Jul 2020 12:05:51 GMT
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.execchain.MainClientExec - Connection can be kept alive indefinitely
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.auth.HttpAuthenticator - Authentication required
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.auth.HttpAuthenticator - my.ssc.portal:443 requested authentication
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.client.TargetAuthenticationStrategy - Authentication schemes in the order of preference: [Negotiate, Kerberos, NTLM, CredSSP, Digest, Basic]
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.client.TargetAuthenticationStrategy - Challenge for Negotiate authentication scheme not available
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.client.TargetAuthenticationStrategy - Challenge for Kerberos authentication scheme not available
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.client.TargetAuthenticationStrategy - Challenge for NTLM authentication scheme not available
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.client.TargetAuthenticationStrategy - Challenge for CredSSP authentication scheme not available
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.client.TargetAuthenticationStrategy - Challenge for Digest authentication scheme not available
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.conn.PoolingHttpClientConnectionManager - Connection [id: 293][route: {s}->https://my.ssc.portal:443] can be kept alive indefinitely
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.conn.DefaultManagedHttpClientConnection - http-outgoing-293: set socket timeout to 0
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.conn.PoolingHttpClientConnectionManager - Connection released: [id: 293][route: {s}->https://my.ssc.portal:443][total kept alive: 1; route allocated: 1 of 20; total allocated: 1 of 200]
14:05:51.111 [Alpine-EventService-7] DEBUG org.apache.http.impl.execchain.MainClientExec - Cancelling request execution
14:05:51.111 [Alpine-EventService-7] WARN org.dependencytrack.integrations.fortifyssc.FortifySscClient - Fortify SSC Client did not receive expected response while attempting to generate a one-time-use fileupload token. HTTP response code: 401 -
14:05:51.111 [Alpine-EventService-7] ERROR org.dependencytrack.integrations.fortifyssc.FortifySscClient - An error occurred while communicating with the Fortify SSC integration point
14:05:51.111 [Alpine-EventService-7] ERROR org.dependencytrack.integrations.fortifyssc.FortifySscClient - HTTP Status : 401
14:05:51.111 [Alpine-EventService-7] ERROR org.dependencytrack.integrations.fortifyssc.FortifySscClient - Request URL : https://my.ssc.portal/api/v1/fileTokens

**Reply by murat-kaya** [reply]
Hi @chrissike, how did you able to get the debug information? Thanks.

Upd:
Found it on doc thanks:
For Docker deployments, simply set the LOGGING_LEVEL environment variable to one of INFO, WARN, ERROR, DEBUG, or TRACE.

**Comment by stevespringett** [comment]
No, it's really just the password. Prior to 20.1, this is what the SSC UI actually did when uploading an FPR. You'll just need to ensure that the user uploading the FPR has the necessary permissions for the project to do so.

**Comment by chrissike** [comment]
Hi @stevespringett , thanks for the quick reply! It seems that a special character in the password had caused the issue. Now the upload is working fine 👍

**Comment by hdshiva** [comment]
Hi @stevespringett I'm using ssc version 18.20.1071.  I used just password to configure ssc in dependencytracker.
I've been seeing the below error when connecting. dependenencytracker running in container.  The user has `security Lead` role to upload. could you please advise ?  Thanks

`01:51:42.940 WARN [FortifySscClient] Fortify SSC Client did not receive expected response while attempting to upload Dependency-Track findings. HTTP response code: 500 - 
01:51:42.941 ERROR [FortifySscClient] An error occurred while communicating with the Fortify SSC integration point
01:51:42.941 ERROR [FortifySscClient] HTTP Status : 500 
01:51:42.941 ERROR [FortifySscClient] Request URL : https://fortify.va.poc.it/ssc/upload/resultFileUpload.html?mat=NjVjZTY4NDEtNWNhNS00MThmLTlhZmEtMWUzMGVmMTIzYzli&engineType=DEPENDENCY_TRACK&entityId=20.7
01:56:42.922 WARN [FortifySscClient] Fortify SSC Client did not receive expected response while attempting to upload Dependency-Track findings. HTTP response code: 500 - 
01:56:42.922 ERROR [FortifySscClient] An error occurred while communicating with the Fortify SSC integration point
01:56:42.922 ERROR [FortifySscClient] HTTP Status : 500 
01:56:42.922 ERROR [FortifySscClient] Request URL : https://fortify.va.poc.it/ssc/upload/resultFileUpload.html?mat=YTY4MzNjN2QtNDgyNy00MGJlLWFkOTQtNjYyODQwZDMwMmY0&engineType=DEPENDENCY_TRACK&entityId=20.7
`

**Comment by hdshiva** [comment]
Upload to ssc worked fine. 👍 
The project configuration's property value attribute was wrong.
